### PR TITLE
Add documentation for building Logstash custom image with opensearch plugin

### PR DESCRIPTION
**Pull request by FloMedja**
### Description

This MR add a documentation to build Logstash custom image with opensearch plugin/

### Issues Resolved

Follow-up to this question #209 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has documentation added
- [x] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: docs/build_custom_logstash_image.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Build A Custom Image with Logstash
```
Comment (dblock) [inline docs/build_custom_logstash_image.md:1]: Capitalize/verbalize consistently, maybe "Building a Custom Docker Image with Logstash"

File: docs/build_custom_logstash_image.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+RUN logstash-plugin install --version 2.0.1 logstash-output-opensearch
+```
+
+## Logstash 7.x
+
+For **`Logstash 7.x`** , the `logstash output opensearch gem` needs to be build.
+
+### Building logstash output opensearch gem plugin
+
+1. Clone `logstash-output-opensearch repo` for the plugin version we want to build. For exemple , clone the  [v2.0.2](https://github.com/opensearch-project/logstash-output-opensearch/tree/2.0.2)
```
Comment (dblock) [inline docs/build_custom_logstash_image.md:23]: Fix spelling, "example", remove extra space before ", "

File: docs/build_custom_logstash_image.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
+# Build A Custom Image with Logstash
+
+If you are looking for a logstash image version with opensearch plugin not available in the [official Docker repository tags](https://hub.docker.com/r/opensearchproject/logstash-oss-with-opensearch-output-plugin/tags) or if you want to include other plugins in your image, it is possible to build a custom image.
+
+## Logstash 8.x
+
+Below is an example of a `Dockerfile` to install `Opensearch 2.0.2` for **`Logstash 8.x`**
```
Comment (dblock) [inline docs/build_custom_logstash_image.md:7]: So one has to create this Dockerfile? Say that.

File: docs/build_custom_logstash_image.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+### Building logstash output opensearch gem plugin
+
+1. Clone `logstash-output-opensearch repo` for the plugin version we want to build. For exemple , clone the  [v2.0.2](https://github.com/opensearch-project/logstash-output-opensearch/tree/2.0.2) 
+
+2. Remove [this line that adds the json version spec](https://github.com/opensearch-project/logstash-output-opensearch/blob/2.0.2/logstash-output-opensearch.gemspec#L49) 
+
+3. Build the gem by running the following command:
+
+    ```bash
```
Comment (dblock) [inline docs/build_custom_logstash_image.md:29]: Align left, add a space after $.

**Review by dblock — CHANGES_REQUESTED** [review]
It's a good start, but it's confusing. Do I have to create the Dockerfile first, but then there are two Dockerfiles in the example? Try to really turn it into a series of steps that one can copy-paste exactly.

+ some nits

File: docs/build_custom_logstash_image.md [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        $gem build logstash-output-opensearch.gemspec
+    ```
+
+It will generate for the gemfile `logstash-output-opensearch-2.0.2-x86_64-linux.gem`
```
[the remark this answers is not in the document]
Comment (FloMedja) [reply docs/build_custom_logstash_image.md:33]: It create it at the root of the repo. I tried the command again.

File: docs/build_custom_logstash_image.md (hunk 4 above)
Comment (FloMedja) [reply docs/build_custom_logstash_image.md:29]: Thank you. I will standardize it to match the format used in other documentation files within the repository.

File: README.md [hunk 6]
```diff
@@ -28,6 +28,7 @@ The logstash-output-opensearch plugin helps to ship events from Logstash to Open
 * [Release Management](RELEASING.md)
 * [Admin Responsibilities](ADMINS.md)
 * [Security](SECURITY.md)
+* [Build custom logstash image](docs/build_custom_logstash_image.md)
```
[the remark this answers is not in the document]
Comment (FloMedja) [reply README.md:31]: Good question... Having a section in the `developer_guide` with a link to the documentation sounds like a good approach to me. @oeyh, what are your thoughts on this?

[the remark this answers is not in the document]
Comment (oeyh) [reply README.md:31]: Sounds good to me.

File: docs/build_custom_logstash_image.md (hunk 1 above)
Comment (FloMedja) [reply docs/build_custom_logstash_image.md]: Done

File: README.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (FloMedja) [reply README.md]: Done

File: docs/build_custom_logstash_image.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
+# Build A Custom Image with Logstash
+
+If you are looking for a logstash image version with opensearch plugin not available in the [official Docker repository tags](https://hub.docker.com/r/opensearchproject/logstash-oss-with-opensearch-output-plugin/tags) or if you want to include other plugins in your image, it is possible to build a custom image.
```
[the remark this answers is not in the document]
Comment (FloMedja) [reply docs/build_custom_logstash_image.md]: Done

File: docs/build_custom_logstash_image.md (hunk 2 above)
Comment (FloMedja) [reply docs/build_custom_logstash_image.md]: Done

File: docs/build_custom_logstash_image.md (hunk 3 above)
Comment (FloMedja) [reply docs/build_custom_logstash_image.md]: Done

File: docs/build_custom_logstash_image.md (hunk 4 above)
Comment (FloMedja) [reply docs/build_custom_logstash_image.md]: Done

**Comment by FloMedja** [comment]
@dblock @oeyh I integrate the comments and rebase. Thanks for taking another look.

**Review by oeyh — APPROVED** [review]
Thanks @FloMedja!

File: docs/dockerfile_to_build_custom_logstash_image.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
+# Dockerfile to build a custom Logstash image
```
Comment (dblock) [inline docs/dockerfile_to_build_custom_logstash_image.md:1]: Building Custom Docker Images

Filename can just be `docker.md`.

File: docs/dockerfile_to_build_custom_logstash_image.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+To build an image that is not available in the [official Docker repository tags](https://hub.docker.com/r/opensearchproject/logstash-oss-with-opensearch-output-plugin/tags), or to add specific plugins to your image, you can create a custom Dockerfile.
+
+The process varies depending on whether you want to build an image with `Logstash 8.x` versions or `Logstash 7.x` versions
+
+## Logstash 8.x
+
+Create this `Dockerfile` to build an image with `Opensearch 2.0.2` for **`Logstash 8.x`**
```
Comment (dblock) [inline docs/dockerfile_to_build_custom_logstash_image.md:9]: Unlike Elasticsearch, correct spelling is OpenSearch, and no need to quote since that's note code.

"Create a `Dockerfile` to build an image with OpenSearch 2.0.2 and Logstash 8.x."

File: docs/dockerfile_to_build_custom_logstash_image.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+    ```
+
+2. Checkout the tag for the plugin version you want to build. For the version [2.0.2](https://github.com/opensearch-project/logstash-output-opensearch/tree/2.0.2) for example
+
+    ```sh
+    git checkout 2.0.2
+    ```
+
+
+3. Remove [this line that adds the json version spec](https://github.com/opensearch-project/logstash-output-opensearch/blob/2.0.2/logstash-output-opensearch.gemspec#L49)
```
Comment (dblock) [inline docs/dockerfile_to_build_custom_logstash_image.md:38]: Btw, why is this needed? Let's explain here.

File: docs/dockerfile_to_build_custom_logstash_image.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+    ```sh
+    gem build logstash-output-opensearch.gemspec
+    ```
+
+    The gemfile `logstash-output-opensearch-2.0.2-x86_64-linux.gem` will be generated.
+
+### Dockerfile
+
+Create this Dockerfile to build an image with `logstash version 7.x` and the previously generated `gemfile`:
```
Comment (dblock) [inline docs/dockerfile_to_build_custom_logstash_image.md:50]: Only quote code. 

gemfile is `Gemfile`

**Review by dblock — CHANGES_REQUESTED** [review]
The spelling of OpenSearch is a must have, everything else up to you. Will merge after.

File: DEVELOPER_GUIDE.md [hunk 12]
```diff
@@ -9,6 +9,7 @@
         - [Configuration for Logstash Output OpenSearch Plugin](#configuration-for-logstash-output-opensearch-plugin)
     - [Submitting Changes](#submitting-changes)
     - [Backports](#backports) 
+    - [Dockerfile to build a custom Logstash image](docs/dockerfile_to_build_custom_logstash_image.md)
```
[the remark this answers is not in the document]
Comment (FloMedja) [reply DEVELOPER_GUIDE.md]: done

File: DEVELOPER_GUIDE.md [hunk 13]
```diff
@@ -215,3 +216,6 @@ original PR with an appropriate label `backport <backport-branch-name>` is merge
 run successfully on the PR. For example, if a PR on main needs to be backported to `1.x` branch, add a label
 `backport 1.x` to the PR and make sure the backport workflow runs on the PR along with other checks. Once this PR is
 merged to main, the workflow will create a backport PR to the `1.x` branch.
+
+## [Dockerfile to build a custom Logstash image](docs/dockerfile_to_build_custom_logstash_image.md)
```
[the remark this answers is not in the document]
Comment (FloMedja) [reply DEVELOPER_GUIDE.md]: done

File: docs/dockerfile_to_build_custom_logstash_image.md (hunk 8 above)
Comment (FloMedja) [reply docs/dockerfile_to_build_custom_logstash_image.md]: done

File: docs/dockerfile_to_build_custom_logstash_image.md (hunk 9 above)
Comment (FloMedja) [reply docs/dockerfile_to_build_custom_logstash_image.md]: Oh, yes. Done.

File: docs/dockerfile_to_build_custom_logstash_image.md (hunk 10 above)
Comment (FloMedja) [reply docs/dockerfile_to_build_custom_logstash_image.md]: Added a little explanation.

File: docs/dockerfile_to_build_custom_logstash_image.md (hunk 11 above)
Comment (FloMedja) [reply docs/dockerfile_to_build_custom_logstash_image.md]: done

**Comment by FloMedja** [comment]
@dblock I integrate the changes and generate a doctree. Thanks for the review :)

**Review by dblock — APPROVED** [review]

**Comment by dblock** [comment]
Thanks! 

Would love more help on this project. There's a bunch of low hanging fruit documentation tasks like https://github.com/opensearch-project/logstash-output-opensearch/issues/234 or https://github.com/opensearch-project/logstash-output-opensearch/issues/141, but also plenty of meatier ones.
